spacy_llm/tasks/entity_linker/registry.py

```diff
@@ -0,0 +1,118 @@
+                    f"vs. {len(example.solutions)}). There must be one reason per solution for an entity "
+                    f"linking example, or no reasons at all. Ignoring all specified reasons."
+                )
+                example.reasons = [""] * len(example.solutions)
+
+    return EntityLinkerTask(
+        template=template,
+        parse_responses=parse_responses or parse_responses_v1,
+        prompt_example_type=example_type,
+        prompt_examples=examples,
+        scorer=scorer or score,
+    )
+
+
+@registry.llm_misc("spacy.CandidateSelector.v1")
+def make_candidate_selector_pipeline(
+    kb_loader: InMemoryLookupKBLoader,
+    top_n: int = 5,
+) -> PipelineCandidateSelector:
+    """Generates CandidateSelector. Note that this class has to be initialized (.initialize()) before being used.
+    kb_loader (InMemoryLookupKBLoader): KB loader.
+    top_n (int): Top n candidates to include in prompt.
+    """
+    # Note: we could also move the class implementation here directly. This was just done to separate registration from
+    # implementation code.
+    return PipelineCandidateSelector(
+        kb_loader=kb_loader,
+        top_n=top_n,
+    )
+
+
+@registry.llm_misc("spacy.EntityDescriptionReader.v1")
+def make_ent_desc_reader() -> EntDescReader:
+    """Instantiates entity description reader with two columns: ID and description.
+    RETURNS (Dict[str, str]): Dict with ID -> description.
+    """
+    return ent_desc_reader_csv
+
+
+@registry.llm_misc("spacy.KBSerializedLoader.v1")
+def make_kb_serialized_loader(
+    path: Union[str, Path],
+    nlp_path: Optional[Union[str, Path]] = None,
+    desc_path: Optional[Union[str, Path]] = None,
+    ent_desc_reader: Optional[EntDescReader] = None,
+) -> KBSerializedLoader:
+    """Instantiates KBSerializedLoader for loading KBs from serialized directories (as done during spaCy pipeline
+    serialization).
+    path (Union[str, Path]): Path to KB directory.
+    nlp_path (Optional[Union[str, Path]]): Path to NLP pipeline whose vocab data to use. If this is None, the loader
+        will try to load the serialized pipeline surrounding the KB directory.
+    desc_path (Optional[Union[Path, str]]): Path to .csv file with descriptions for entities. Has to have two
+        columns with the first one being the entity ID, the second one being the description. The entity ID has to
+        match with the entity ID in the stored knowledge base.
+        If not specified, all entity descriptions provided in prompts will be a generic "No description available"
+        or something else to this effect.
+    ent_desc_reader (Optional[EntDescReader]): Entity description reader.
+    RETURNS (KBSerializedLoader): Loader instance.
+    """
+    return KBSerializedLoader(
+        path=path,
+        nlp_path=nlp_path,
+        desc_path=desc_path,
+        ent_desc_reader=ent_desc_reader or ent_desc_reader_csv,
+    )
+
+
+@registry.llm_misc("spacy.KBYamlLoader.v1")
```

Comment:
Could we call this more generically `KBFileLoader`, and have v1 only support yaml?

Reply:
Either way, the current name & capitalization is really hard to read, we'd need to brainstorm some alternatives.

Reply:
Done in https://github.com/explosion/spacy-llm/commit/2e2895c1441c6336acfb852bfb31d5be1e5dfd65.